server/go/subscription/watcher.go

```diff
@@ -0,0 +1,28 @@
+package subscription
+
+import "github.com/google/uuid"
+
+type Subscriber interface {
+	ID() string
+	Notify(message *Message) error
```

Comment:
I think it's really dangerous to have the message as a pointer. It makes the message mutable and thread unsafe. Is it really neccesary?

Reply:
Very good point.